frontend/www/js/omegaup/api.js

```diff
@@ -635,9 +636,23 @@ export default {
 
     listAssociatedIdentities: _call('/api/user/listAssociatedIdentities/'),
 
-    listUnsolvedProblems: _call('/api/user/listUnsolvedProblems/'),
+    listUnsolvedProblems: _call('/api/user/listUnsolvedProblems/', function(
+      data,
+    ) {
+      if (!data.hasOwnProperty('problems')) return data;
+      data.problems = data.problems.map(problem => {
```

Comment:
nit: si el lambda es una sóla línea que regresa un valor, puedes reemplazarlo por

```suggestion
      data.problems = data.problems.map(
        problem => new Problem(problem)
      );
```